### PR TITLE
waterfall plot: use display_data if exists

### PR DESCRIPTION
**Pull request by yuuuxt**
not sure if this is a complete fix and if it breaks anything. Please review, thanks!

### CODE REVIEW

File: shap/_explanation.py [hunk 1]
```diff
@@ -323,9 +323,10 @@ def __getitem__(self, item):
             pos += 1
 
             # skip over Ellipsis
-            if t == Ellipsis:
-                pos += len(self.shape) - len(item)
-                continue
+            if type(t) not in [np.ndarray, pd.Series]:
```
Comment (slundberg) [inline shap/_explanation.py]: Can you elaborate on scenarios where this change is needed? Thanks!

**Comment by slundberg** [comment]
Thanks for the fix! Sorry I was slow on the review (was out for a while after paternity leave). See the one comment I left.

File: shap/_explanation.py (hunk 1 above)
Comment (yuuuxt) [reply shap/_explanation.py]: oh, that was for #2325 , just showing my naive "fix" for that issue. Now it's reverted, so it's clean for this PR now.

**Comment by yuuuxt** [comment]
> Thanks for the fix! Sorry I was slow on the review (was out for a while after paternity leave). See the one comment I left.

Fixed that (replied above).

And congrats! :)

**Pull request merged**